controller/composite/controller.go

```diff
@@ -536,7 +536,7 @@ func (pc *parentController) updateParentStatus(parent *unstructured.Unstructured
 	// We can't use Patch() because we need to ensure that the UID matches.
 	// TODO(enisoc): Use /status subresource when that exists.
 	// TODO(enisoc): Update status.observedGeneration when spec.generation starts working.
-	nsParentClient := pc.parentClient.WithNamespace(parent.GetNamespace())
+	nsParentClient := pc.parentClient
```

Comment:
Simmarly to above about calling `pc.parentClient.UpdateWIthRetries()...`  directly instead of defining a new variable. We could use:

```
return pc.parentClient.UpdateWithRetries(...
```

Reply:
+1, you are right. :)

Reply:
Also like above, this may need to change to account for `pc.parentClient` becoming namespace-agnostic. To make something like `pc.parentClient.Namespace(...).UpdateWithRetries(...)` work, we'll need to restructure how we add extra methods to ResourceInterface in our `dynamic/clientset` wrapper.

Reply:
Thanks for the detailed explanation. Did you mean `ResourceClient`? `NamespaceableResourceInterface` is a vendor API.

Reply:
The original purpose of ResourceClient was to extend ResourceInterface with extra methods that are useful for Metacontroller, which works with many different resources from many different angles.

Now that the new dynamic client has been split into ResourceInterface and NamespaceableResourceInterface, we need to restructure ResourceClient to correspond to NamespaceableResourceInterface, and add another level of wrapper to extend ResourceInterface as well.